packages/syft/src/syft/store/blob_storage/__init__.py

```diff
@@ -135,13 +136,13 @@ def syft_iter_content(
                 ):
                     current_byte += len(chunk)
                     yield chunk
-                return
-
+            return  # If successful, exit the function
         except requests.exceptions.RequestException as e:
-            if attempt < max_retries:
+            if attempt < max_retries - 1:
                 print(
-                    f"Attempt {attempt}/{max_retries} failed: {e} at byte {current_byte}. Retrying..."
+                    f"Attempt {attempt + 1}/{max_retries} failed: {e} at byte {current_byte}. Retrying..."
                 )
+                time.sleep(2**attempt)  # exponential backoff
```

Comment:
I think this is a potential change but let's make this exponential backoff into a separate PR.

Reply:
https://github.com/OpenMined/PySyft/pull/8900